### PR TITLE
opensource files and automation

### PR DESCRIPTION
**Pull request by onyxraven**
Add travis.yml and our opensource fileset.

https://github.com/aws/aws-xray-daemon

The xray daemon is under the Apache-2 license so can use the same.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+services:
+  - docker
+before_script:
+  - docker login -u="$DOCKER_USERNAME" -p="$DOCKER_PASSWORD"
```
Comment (awwithro) [inline .travis.yml]: do these vars need to be set somewhere?

Comment (onyxraven) [reply .travis.yml]: I've got them setup in the travis UI.  thats how I setup the other one.

File: Dockerfile [hunk 2]
```diff
@@ -1,16 +1,12 @@
-FROM debian:stable-slim
+FROM alpine as download
+
+ADD https://s3.dualstack.us-east-2.amazonaws.com/aws-xray-assets.us-east-2/xray-daemon/aws-xray-daemon-linux-2.x.zip /tmp
+RUN cd /tmp/ && apk add --update libc6-compat unzip && unzip aws-xray-daemon-linux-2.x.zip && mv /tmp/xray /usr/bin/ && rm /tmp/aws-xray-daemon-linux-2.x.zip
```
Comment (onyxraven) [inline Dockerfile]: switched to alpine. xray needs libc6 compatibility. built image is 18mb (ubuntu img is 42mb)

File: Dockerfile [hunk 3]
```diff
@@ -1,16 +1,15 @@
-FROM debian:stable-slim
+FROM alpine as download
+
+ADD https://s3.dualstack.us-east-2.amazonaws.com/aws-xray-assets.us-east-2/xray-daemon/aws-xray-daemon-linux-2.x.zip /tmp
+RUN cd /tmp/ && apk add --update libc6-compat unzip && \
+		unzip aws-xray-daemon-linux-2.x.zip && mv /tmp/xray /usr/bin/ && \
+		rm /tmp/* && \
+		rm -rf /var/cache/apk/*
```
Comment (onyxraven) [inline Dockerfile]: switched to alpine. xray needs libc6 compatibility. built image is 14.5mb (ubuntu img is 42mb)

**Pull request merged**